### ISSUE TITLE
HEAD requests not supported

### ISSUE THREAD

**Issue by chris968** [passbolt/passbolt_api#507]
## ISSUE NAME
* Passbolt Version: 4.5.2 (CE)
* Platform and Target:
-- Operating system: CentOS 7
-- PHP: 7.4
-- Web server: nginx
-- Database server: mariadb

HEAD requests are deactivated on all routes, some monitoring platform only support HEAD request (this might depend on subscription level). 
It could be interesting to support HEAD requests, one some routes at least, if there is no security concern of course.

**Comment by stripthis** [comment]
Hello, Can you give us an example of such monitoring platform? Are you doing monitoring on client side or server side? Which API endpoints you would need in particular? Currently we have a `GET /healthcheck/status.json` that can be used by probes, so we could add HEAD for this.

**Comment by chris968** [comment]
I use free UptimeRobot subscription and it only support HEAD requests which is lead to a 404 error on any path I've tested. (`/`, `/healthcheck/status.json`, `/healthcheck/status`).
I manually modified the `routes.php` file and it worked as expected.

```
/**
 * Healthchecks routes
 */
$routes->scope('/healthcheck', function ($routes) {
    $routes->setExtensions(['json']);

    $routes->connect('/status', ['prefix' => 'Healthcheck', 'controller' => 'HealthcheckStatus', 'action' => 'status'])
        ->setMethods(['GET','HEAD']);

    $routes->connect('/', ['prefix' => 'Healthcheck', 'controller' => 'HealthcheckIndex', 'action' => 'index'])
        ->setMethods(['GET']);

    $routes->connect('/error', ['prefix' => 'Healthcheck', 'controller' => 'HealthcheckError', 'action' => 'internal'])
        ->setMethods(['GET','POST', 'PUT', 'DELETE']);
});
```

**Comment by ishanvyas22** [comment]
Hey @chris968, support for HEAD method is released with [v4.7](https://github.com/passbolt/passbolt_api/releases/tag/v4.7.0). Feel free to try it out and thank you for your patience.
